File: ARRAYS/Q1. CAN PLACE FLOWERS/fatimahadnan230--Q1. Can Place Flowers.cpp
#include <vector>
using namespace std;

class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int count = 0;
        int size = flowerbed.size();
        
        for (int i = 0; i < size; i++) {
            // check if current plot is empty
            if (flowerbed[i] == 0) {
                // check if left and right are empty (or at boundaries)
                bool emptyLeft = (i == 0) || (flowerbed[i - 1] == 0);
                bool emptyRight = (i == size - 1) || (flowerbed[i + 1] == 0);
                
                if (emptyLeft && emptyRight) {
                    flowerbed[i] = 1; // plant it
                    count++;
                    
                    if (count >= n) return true;
                }
            }
        }
        return count >= n;
    }
};

File: ARRAYS/Q1. CAN PLACE FLOWERS/fatimahadnan230--Q1. Can Place Flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        count = 0
        length = len(flowerbed)
        
        for i in range(length):
            # check if the current plot is empty
            if flowerbed[i] == 0:
                # check if left and right neighbors are empty or out of bounds
                empty_left = (i == 0) or (flowerbed[i - 1] == 0)
                empty_right = (i == length - 1) or (flowerbed[i + 1] == 0)
                
                if empty_left and empty_right:
                    # plant a flower here
                    flowerbed[i] = 1
                    count += 1
                    # if we've already planted enough, we can stop early
                    if count >= n:
                        return True
                        
        return count >= n